sp/src/game/client/c_basecombatweapon.cpp

```diff
@@ -551,6 +561,24 @@ int C_BaseCombatWeapon::DrawModel( int flags )
 	return BaseClass::DrawModel( flags );
 }
 
+#ifdef MAPBASE
+//-----------------------------------------------------------------------------
+// Purpose:	
+//-----------------------------------------------------------------------------
+bool C_BaseCombatWeapon::DispatchMuzzleEffect( const char *options, bool isFirstPerson )
+{
+	// Don't show muzzle flashes in first-person
+	C_BasePlayer *pPlayer = ToBasePlayer( GetOwner() );
+	if (pPlayer)
+	{
+		if (pPlayer->DrawingPlayerModelExternally() && pPlayer->InFirstPersonView())
+			return false;
```

Comment:
Does this change the behavior of muzzle flashes in EZ2? I know that in EZ and EZ2 muzzle flashes create a d-light to light up the surroundings and I think this applies to the player as well. I also remember working on projected texture based muzzle flashing lighting for the player but I cannot remember if we finished that or not.